### ISSUE TITLE
[RFC]: Support Qwen3-Omni in SGL_JAX

### ISSUE THREAD

**Issue by Garrybest** [sgl-project/sglang-jax#703]
# 1. Abstract
Now sglang-jax framework has supported multimodal model according to #552. This proposal will introduce and support a new multimodal model Qwen3-Omni.
# 2. Motivation
The Qwen3-Omni model, a recent multimodal large language model from Alibaba, demonstrates strong capabilities in handling diverse inputs including text, audio, and vision within a unified architecture. Integrating Qwen3-Omni into sglang-jax would enable users to leverage its advanced multimodal reasoning within the high-performance, JAX-based inference framework that sglang-jax provides.
Meanwhile, Qwen3-Omni has a complex model structure and data flow. Supporting this model would help us improve the sglang-jax framework to handle multimodal input.

<img width="2688" height="2809" alt="Image" src="https://github.com/user-attachments/assets/0dafe3a3-c4b7-4ced-9ed5-c7874bcecd4f" />

# 3. Architecture Design

**Tokenizer**：
- Fetch and load raw multimodal input
  - audio: fetch and load(by librosa) -> np.ndarray
  - image: fetch and parse -> Image.Image
  - video: fetch and parse(by opencv) -> np.ndarray
- Process multimodal items to get input multimodal features
  - implement Qwen3OmniMoeProcessor
    - audio: WhisperFeatureExtractor
    - image: Qwen2VLImageProcessor
    - video: Qwen2VLVideoProcessor
    - text: 
      - extend text by multimodal pad tokens, based on conversation chat template and video/audio timeline
      - tokenize text
  - output: input_ids, mm_items, mrope_positions

**Model**:
- **Stage 0**: Qwen3OmniMoeEncoder
  - embedding multimodal features
    - audio: Qwen3OmniMoeAudioEncoder for audio features
    - visual: Qwen3OmniMoeVisionEncoder for image and video features, return features and deepstack_visual_embeds
  - text:
    - embedding for text tokens
    - Integrate multimodal embeddings with text token embeddings by placeholders
- **Stage 1**: Qwen3OmniMoeThinkerForConditionalGeneration
  - leverage Auto-Regressive(AR) scheduler to generate tokens
  - Generate as Auto-Regressive(AR) Model
  - Global Scheduler receives output tokens and hidden states from Thinker.
- **Stage 2**: Qwen3OmniMoeTalkerForConditionalGeneration(Optional)
  - leverage Auto-Regressive(AR) scheduler to generate tokens
  - prepare inputs
    - prefill: prepare talker_input_embed and trailing_text_hidden from user and assistant embedding part
    - decode: implement code predictor model to get input residuals and embeddings, code predictor is an fixed-step Auto-Regressive(AR) generation model.
  - Generate as Auto-Regressive(AR) Model
  - Global Scheduler receives output hidden states from Talker.
- **Stage 3**: Qwen3OmniMoeCode2Wav(Optional)
  - Forward by chunks(Talker output hidden_states)

<img width="1841" height="925" alt="Image" src="https://github.com/user-attachments/assets/56828ee6-354a-4a5d-9e68-49cd58d6e02e" />

# 4. Inter-stage Data Definition
```python
Stage 0:
    # text
    input_ids: jax.Array
    attention_mask: jax.Array | None = None

    # image
    pixel_values: jax.Array | None = None
    image_grid_thw: jax.Array | None = None
    
    # video
    pixel_values_videos: jax.Array | None = None
    video_grid_thw: jax.Array | None = None
    video_second_per_grid: jax.Array | None = None
    
    # audio
    input_features: jax.Array | None = None
    feature_attention_mask: jax.Array | None = None

Stage 1:
    # from stage 0
    thinker_inputs_embeds: jax.Array
    visual_pos_masks: jax.Array | None = None
    deepstack_visual_embeds: jax.Array | None = None

    # from input
    attention_mask: jax.Array | None = None

Stage 2:
    # from stage 1
    # hidden states
    thinker_hidden_states: jax.Array
    # text output ids from thinker including prompt ids
    thinker_sequences: jax.Array
    # special token embedding from thinker: tts_bos_embed, tts_eos_embed, tts_pad_embed
    talker_special_token_embeddings: jax.Array 

    # from stage 0
    # deepstack embeddings from ViT
    deepstack_visual_embeds: jax.Array | None = None

Stage 3:
    # from stage 2
    talker_hidden_states: jax.Array
```

# 5. RoadMap(Work Breakdown)

- [x] **Milestone 1**: implement the basic Qwen3-Omni model without audio output
  - [x] Tokenizer
    - [x] Fetch and load raw multimodal input
    - [x] Process multimodal items to get input multimodal features
  - [x] Encoder
    - [x] Qwen3OmniMoeAudioEncoder @Garrybest 
    - [x] Qwen3OmniMoeVisionEncoder(with deepstack features) @SII-limingliu 
    - [x] Embedding tokens by text input_ids and multimodal encoder output @Garrybest 
  - [x] Thinker @pathfinder-pf 
    - [x] Qwen3OmniMoeThinkerTextModel development

- [ ] **Milestone 2**: implement the rest Talker and Code2Wav models to generate audio output
  - [ ] Talker
    - [ ] Handle user and assistant embedding part
    - [ ] Code predictor MTP module development and integration with embedding step
  - [ ] Code2Wav development

- [ ] **Milestone 3**: Support streaming output mode

- [ ] **Other works**:
  - [ ] Unit test / e2e test / add to CI
  - [ ] Model evaluation
